Unable to generate single component in `Procedure_push_docker_images`

### Description

As part of [wazuh/wazuh-qa-automation#6711](https://github.com/wazuh/wazuh-qa-automation/issues/6711), the package generation workflow is being split so that each Docker component (`wazuh-manager`, `wazuh-indexer`, `wazuh-dashboard`, `wazuh-agent`) is dispatched independently instead of all four at once. This allows each component to fail and be retried in isolation, without blocking the others.

When Procedure_push_docker_images.yml is triggered with a single component using the products input (e.g. wazuh-manager), the workflow fails in the generate package urls step.

When `Procedure_push_docker_images.yml` is dispatched with a single component via the `products` input (e.g. `products: wazuh-manager`), the `generate package urls` job fails because the script `generate_presigned_dev_urls.py` is called with empty revision arguments for the components not included in the dispatch.

**Failing run:** https://github.com/wazuh/wazuh-docker/actions/runs/24888300685

**Error:**
```console
Run source /home/runner/work/wazuh-docker/wazuh-docker/workflow_venv/bin/activate
usage: generate_presigned_dev_urls.py [-h] --process PROCESS --wazuh-version
                                      WAZUH_VERSION --aws-s3-bucket-dev
                                      AWS_S3_BUCKET_DEV
                                      [--indexer-revision INDEXER_REVISION]
                                      [--manager-revision MANAGER_REVISION]
                                      [--dashboard-revision DASHBOARD_REVISION]
                                      [--agent-revision AGENT_REVISION]
                                      [--ova-revision OVA_REVISION]
                                      [--assistant-revision ASSISTANT_REVISION]
                                      [--expires-in EXPIRES_IN]
generate_presigned_dev_urls.py: error: argument --indexer-revision: expected one argument
Error: Process completed with exit code 2.
```

--- a/.github/workflows/Procedure_push_docker_images.yml
+++ b/.github/workflows/Procedure_push_docker_images.yml
@@ -233,6 +233,11 @@ jobs:
           source ${{ env.WORKFLOW_VENV }}/bin/activate
           WAZUH_COMPONENTS='${{ needs.setup.outputs.WAZUH_COMPONENTS }}'
           COMMIT_LIST='${{ needs.setup.outputs.COMMIT_LIST }}'
+          SCRIPT_PARAMS="--process ${{ env.PRESIGNED_URLS_SCRIPT_PROCESS }} \
+                            --wazuh-version ${{ env.WAZUH_VERSION }} \
+                            --aws-s3-bucket-dev ${{ env.LOCAL_AWS_S3_BUCKET_DEV }} \
+                            --assistant-revision $ASSISTANT_REVISION "
+
 
           # Parse components and their revisions
           COMPONENTS=($(echo "$WAZUH_COMPONENTS" | jq -r '.[]'))
@@ -251,29 +256,22 @@ jobs:
           for i in "${!COMPONENTS[@]}"; do
           case "${COMPONENTS[$i]}" in
             wazuh-manager)
-            MANAGER_REVISION="${REVISIONS[$i]}"
+            SCRIPT_PARAMS+="--manager-revision ${REVISIONS[$i]} "
             ;;
             wazuh-dashboard)
-            DASHBOARD_REVISION="${REVISIONS[$i]}"
+            SCRIPT_PARAMS+="--dashboard-revision ${REVISIONS[$i]} "
             ;;
             wazuh-indexer)
-            INDEXER_REVISION="${REVISIONS[$i]}"
+            SCRIPT_PARAMS+="--indexer-revision ${REVISIONS[$i]} "
             ;;
             wazuh-agent)
-            AGENT_REVISION="${REVISIONS[$i]}"
+            SCRIPT_PARAMS+="--agent-revision ${REVISIONS[$i]} "
             ;;
           esac
           done
 
           python ${{ env.GENERATE_PRESIGNED_URLS_SCRIPT_PATH }} \
-              --process ${{ env.PRESIGNED_URLS_SCRIPT_PROCESS }} \
-              --wazuh-version ${{ env.WAZUH_VERSION }} \
-              --aws-s3-bucket-dev ${{ env.LOCAL_AWS_S3_BUCKET_DEV }} \
-              --indexer-revision   $INDEXER_REVISION \
-              --manager-revision   $MANAGER_REVISION \
-              --dashboard-revision $DASHBOARD_REVISION \
-              --agent-revision     $AGENT_REVISION \
-              --assistant-revision $ASSISTANT_REVISION
+              $SCRIPT_PARAMS
 
       - name: Save presigned URLs file to artifact
         if: ${{ inputs.dev == true }}

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ All notable changes to this project will be documented in this file.
 
 ### Changed
 
+- Presigned URLs generation enhancement ([#2346](https://github.com/wazuh/wazuh-docker/pull/2346))
 - Adapt bumper workflows to change main branch ([#2294](https://github.com/wazuh/wazuh-docker/pull/2294))
 - Delete all API user and password references and Wazuh agent references ([#2289](https://github.com/wazuh/wazuh-docker/pull/2289))
 - Create certificate directory with default user and group ([#2287](https://github.com/wazuh/wazuh-docker/pull/2287))

